### ISSUE TITLE
Try using Material-UI to spruce up the UI

### ISSUE THREAD

**Issue by schue** [filterbubbler/filterbubbler-web-ext#5]
We don't want to develop our own UI toolkit from scratch to get a good looking WebExtension UI. The Material UI project provides a set of React components that implement the Google Material UI look and feel specification (http://www.material-ui.com/) and that can provide a big boost in reaching our goal. We need to test how well this library works in the WebExtensions context.

**Comment by schue** [comment]
I have successfully embedded Material-UI into the system. The complication is that Material-UI expects to manipulate the application state in the manner described by the "Flux" architecture (https://facebook.github.io/flux/docs/overview.html). This means we need an object modeling the application's state that the UI can bind to. Since the state of the WebExtension pop-up is reconstructed every time the user selects the browser action button we need to send the state object from the background script whenever that occurs and then forward changes to the background whenever the user changes any options. Happily there is a project that already attempts to provide this functionality (https://github.com/ivantsov/redux-webext) and seems, at first blush, to work fairly well. With that functionality in place we then need some tools for managing forms and validation. After some digging around it appears that a popular solution is the Redux Form library (http://redux-form.com). I had initial success installing this library and getting animated form fields in place but then things become complicated. Some of the events that are forwarded across the WebExtension redux bridge do not seem to function correctly and end up throwing errors about variables that have not been initialized. In Firefox the line numbers for these errors are not translated by sourcemaps so debugging is greatly complicated. Since I am trying to use ES2015 features via Babel the line translations are pretty much required. After trying a variety of sourcemap settings for webpack unsuccessfully I have decided to try my luck in Chrome and also spend a little time examining the WebExtensions browser portability issues. My hope is that I may kill two birds with one stone by getting the extension running in Chrome and discovering that it can support sourcemaps. If that is the case I can keep moving forward on development while I also try to chase down Firefox developers who are working on sourcemap support in webextensions.
